src/transformer/typeReferenceReusable/typeReferenceReusable.ts

```diff
@@ -1,11 +1,18 @@
-import * as ts from 'typescript';
+import type * as ts from 'typescript';
+import { core } from '../core/core';
 
-const reusableTypes: ts.SyntaxKind[] = [
-  ts.SyntaxKind.ClassDeclaration,
-  ts.SyntaxKind.InterfaceDeclaration,
-  ts.SyntaxKind.TypeAliasDeclaration,
-];
+let reusableTypes: null | ts.SyntaxKind[] = null;
+
+function getReusableTypes(): ts.SyntaxKind[] {
+  return [
+    core.ts.SyntaxKind.ClassDeclaration,
+    core.ts.SyntaxKind.InterfaceDeclaration,
+    core.ts.SyntaxKind.TypeAliasDeclaration,
+  ];
+}
 
 export function isTypeReferenceReusable(declaration: ts.Declaration): boolean {
+  reusableTypes = reusableTypes || getReusableTypes();
```

Comment:
You could just inline the reusableTypes inside the function instead of a method. It's true that your way caches the array... but I don't think there is a performance concern here

Reply:
makes sense, done